src/util/std_types.h

```diff
@@ -768,6 +768,15 @@ class array_typet:public type_with_subtypet
     add(ID_size, std::move(_size));
   }
 
+  /// The type of the index expressions into any instance of this type.
+  typet index_type() const;
+
+  /// The type of the elements of the array.
+  const typet &element_type() const
+  {
+    return subtype();
```

Comment:
Should we actually hide `subtype()` for `array_typet` and `vector_typet`?

Reply:
Yes, that will improve readability.